spring-boot-project/spring-boot-actuator-autoconfigure/src/test/java/org/springframework/boot/actuate/autoconfigure/metrics/test/MetricsRun.java

```diff
@@ -90,7 +91,7 @@ private MetricsRun() {
 	 * implementation.
 	 * @return the function to apply
 	 */
-	public static Function<ApplicationContextRunner, ApplicationContextRunner> simple() {
+	public static <T extends AbstractApplicationContextRunner> Function<T, T> simple() {
```

Comment:
This is not strictly related to the change. I am happy to revist the tests here but I'd like to do that in a separate PR. 

Could you please revert everything that's related to that?

(also if submit a new PR for this, there is a raw type use of `AbstractApplicationContextRunner`